src/Servers/Kestrel/Core/src/Internal/Http3/QPack/QPackEncoder.cs

```diff
@@ -0,0 +1,504 @@
+            {
+                EncodeHeaderBlockPrefix(destination, out bytesWritten);
+                destination = destination.Slice(bytesWritten);
+
+                return IntegerEncoder.Encode(index, 6, destination, out bytesWritten);
+            }
+
+            bytesWritten = 0;
+            return false;
+        }
+
+        public static bool EncodeIndexHeaderFieldWithPostBaseIndex(int index, Span<byte> destination, out int bytesWritten)
+        {
+            bytesWritten = 0;
+            return false;
+        }
+
+        /// <summary>Encodes a "Literal Header Field without Indexing".</summary>
+        public static bool EncodeLiteralHeaderFieldWithNameReference(int index, string value, Span<byte> destination, out int bytesWritten)
+        {
+            if (destination.Length != 0)
+            {
+                EncodeHeaderBlockPrefix(destination, out bytesWritten);
+                destination = destination.Slice(bytesWritten);
+
+                return IntegerEncoder.Encode(index, 6, destination, out bytesWritten);
+            }
+
+            bytesWritten = 0;
+            return false;
+        }
+
+        /*
+         *         0   1   2   3   4   5   6   7
+                  +---+---+---+---+---+---+---+---+
+                  | 0 | 1 | N | S |Name Index (4+)|
+                  +---+---+---+---+---------------+
+                  | H |     Value Length (7+)     |
+                  +---+---------------------------+
+                  |  Value String (Length bytes)  |
+                  +-------------------------------+
+         */
+        public static bool EncodeLiteralHeaderFieldWithPostBaseNameReference(int index, Span<byte> destination, out int bytesWritten)
+        {
+            bytesWritten = 0;
+            return false;
+        }
+
+        public static bool EncodeLiteralHeaderFieldWithoutNameReference(int index, Span<byte> destination, out int bytesWritten)
+        {
+            bytesWritten = 0;
+            return false;
+        }
+
+        /*
+         *     0   1   2   3   4   5   6   7
+               +---+---+---+---+---+---+---+---+
+               |   Required Insert Count (8+)  |
+               +---+---------------------------+
+               | S |      Delta Base (7+)      |
+               +---+---------------------------+
+               |      Compressed Headers     ...
+               +-------------------------------+
+         *
+         */
+        private static bool EncodeHeaderBlockPrefix(Span<byte> destination, out int bytesWritten)
+        {
+            int length;
+            bytesWritten = 0;
+            // Required insert count as first int
+            if (!IntegerEncoder.Encode(0, 8, destination, out length))
+            {
+                return false;
+            }
+
+            bytesWritten += length;
+            destination = destination.Slice(length);
+
+            // Delta base
+            if (destination.Length == 0)
```

Comment:
```suggestion
            if (destination.IsEmpty)
```
elides the bound check (see other comment).